src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -81,9 +81,8 @@ public Task<T> InvokeAsync<T>(string targetName, IEnumerable<object> arguments,
 
         public void Dispose()
         {
-            OnDisposed();
-
-            _rpc.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
This is the correct pattern for all non-sealed types, whether or not they have a finalizer. The reason for this is derived types are allowed to introduce a finalizer, but according to the best practices for finalization those derived types are not allowed to also re-implement `IDisposable.Dispose()` for the purpose of adding the corresponding call to `GC.SuppressFinalize`. The call becomes essentially a no-op for cases where there is no user-defined finalizer, but ensures the code continues to work should one be added.